File: src/LCR046.WNC0Lk.1000299/solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var rightSideView = function (root) {
    const result = [];
    if (!root) {
        return result;
    }
    let nodes = [root, ];
    while (nodes.length) {
        const nNodes = [];
        for (let i = 0; i < nodes.length; i++) {
            nodes[i].left && nNodes.push(nodes[i].left);
            nodes[i].right && nNodes.push(nodes[i].right);
        }
        result.push(nodes[nodes.length - 1].val);
        nodes = nNodes;
    }
    return result;
};

File: src/LCR046.WNC0Lk.1000299/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        if(root == null){
            return List.of();
        }
        List<Integer> result = new ArrayList<>();
        travel(List.of(root),result);
        
        return result;
    }
    
    private void travel(List<TreeNode> level,List<Integer> result){
        int last = -1;
        List<TreeNode> next = new ArrayList<>();
        for(TreeNode node : level ){
            last = node.val;
            if(node.left != null){
                next.add(node.left);
            }
            if(node.right != null){
                next.add(node.right);
            }
        }
        result.add(last);
        if(!next.isEmpty()){
            travel(next,result);
        }
    }
    
}